bskyweb/cmd/bskyweb/server.go

```diff
@@ -342,6 +343,42 @@ func (srv *Server) WebPost(c echo.Context) error {
 		}
 		data["imgThumbUrls"] = thumbUrls
 	}
+
+	// expand links in rich text back to full urls, replacing shortened urls in
+	// social card meta tags and the noscript output. this essentially reverses the effect
+	// of shortenLinks() in src/lib/strings/rich-text-manip.ts
+	if postView.Record != nil {
+		rec := postView.Record.Val.(*appbsky.FeedPost)
+		postText := rec.Text
+		var charsAdded int64 = 0
+		// iterate over facets, check if they're link facets, and if found, grab the uri
+		for _, facet := range rec.Facets {
+			linkUri := ""
+			if slices.ContainsFunc(facet.Features, func(feat *appbsky.RichtextFacet_Features_Elem) bool {
+				if feat.RichtextFacet_Link != nil && feat.RichtextFacet_Link.LexiconTypeID == "app.bsky.richtext.facet#link" {
+					linkUri = feat.RichtextFacet_Link.Uri
+					// only expand uris that have been shortened (as opposed to those with mismatched text/uris)
+					if strings.HasSuffix(postText[facet.Index.ByteStart+charsAdded:facet.Index.ByteEnd+charsAdded], "...") &&
```

Comment:
String indexing like this should work in terms of correct offsets: the facet byte ranges are UTF-8 byte offsets, and that is how string indexing works in golang.

However, there needs to be some bounds checking here, to prevent out-of-bounds index panics.